source/yats/include/yats/task_container.h

```diff
@@ -13,12 +14,23 @@ namespace yats
 class abstract_task_container
 {
 public:
-    abstract_task_container() = default;
+    abstract_task_container(std::set<size_t> following_nodes)
+        : m_following_nodes(std::move(following_nodes))
+    {
+    }
 
     virtual ~abstract_task_container() = default;
 
     virtual void run() = 0;
     virtual bool can_run() const = 0;
+
+    const std::set<size_t>& following_nodes()
```

Comment:
Would be nicer if this was a vector instead of a set, since we do not need any properties of a set at this point.